app/src/main/java/ca/klapstein/baudit/activities/LoginCareProviderActivity.java

```diff
@@ -20,49 +25,70 @@
 
     private LoginPresenter presenter;
 
+    private EditText usernameInput;
+    private EditText passwordInput;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_login_care_provider);
 
         presenter = new LoginPresenter(this);
-    }
 
-    /**
-     * Attempt a login as a {@code CareProvider}.
-     *
-     * @return {@code true} if the login was successful, otherwise return {@code false}.
-     */
-    public boolean login() {
-        // TODO: implement
-        return true;
-    }
+        usernameInput = findViewById(R.id.enter_care_provider_username);
+        passwordInput = findViewById(R.id.enter_care_provider_password);
 
-    @Override
-    public void setUserNameError() {
+        Button loginButton = findViewById(R.id.login_care_provider_button);
+        loginButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.onLoginButtonClicked(
+                    usernameInput.getText().toString(),
+                    passwordInput.getText().toString()
+                );
+            }
+        });
+
+        Button registerButton = findViewById(R.id.register_care_provider_button);
+        registerButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(getBaseContext(), CreateCareProviderAccountActivity.class);
```

Comment:
Is using `getBaseContext()` the best context grabber for this use?

https://stackoverflow.com/questions/10641144/difference-between-getcontext-getapplicationcontext-getbasecontext-and

Should we be inheriting from this activities context, the applications, or another activities context.

Your thoughts @thomaslorincz ?

Reply:
Good point. getApplicationContext is more appropriate.

Reply:
🕵️‍♂️  Scratch that. I'm investigating what is more appropriate...